### ISSUE TITLE
false-positive: assign on dereferenced pointer receiver

### ISSUE THREAD

**Issue by ldez** [sivchari/nilassign#4]
```go
package sandbox

import (
	"strconv"
	"time"
)

type Duration time.Duration

func (d *Duration) Set(s string) error {
	if d == nil {
		return nil
	}

	if v, err := strconv.ParseInt(s, 10, 64); err == nil {
		*d = Duration(time.Duration(v) * time.Second)
		return nil
	}

	v, err := time.ParseDuration(s)
	*d = Duration(v)
	return err
}
```

```console
$ ./golangci-lint run
main.go:16:3: this assignment occurs invalid memory address or nil pointer dereference (nilassign)
                *d = Duration(time.Duration(v) * time.Second)
                ^
main.go:21:2: this assignment occurs invalid memory address or nil pointer dereference (nilassign)
        *d = Duration(v)
        ^
```

**Comment by sivchari** [comment]
This problem is fixed. Thanks

**Comment by sivchari** [comment]
@ldez 
#6  is fixed this and #5

**Comment by sivchari** [comment]
I'm checking other false-positive before pushing new PR to make sure.

**Comment by ldez** [comment]
@sivchari don't close the issue before the fix is published.

You can add this in the description of your PR to close automatically the issues:
```
Fixes #3 
Fixes #4
```

**Comment by ldez** [comment]
Fixed by #6
